hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieCompactedLogRecordScanner.java

```diff
@@ -63,69 +66,95 @@
   private AtomicLong totalLogRecords = new AtomicLong(0);
   // Total final list of compacted/merged records
   private long totalRecordsToUpdate;
+  // Latest valid instant time
+  private String latestInstantTime;
 
   public HoodieCompactedLogRecordScanner(FileSystem fs, List<String> logFilePaths,
-      Schema readerSchema) {
+      Schema readerSchema, String latestInstantTime) {
     this.readerSchema = readerSchema;
+    this.latestInstantTime = latestInstantTime;
 
+    // Store only the last data block (needed to implement rollback)
+    Optional<HoodieLogBlock> lastBlock = Optional.empty();
```

Comment:
Now cache the last LogBlock instead of the last records read, this is done so we can rollback() delete as well.